mesonbuild/compilers/mixins/visualstudio.py

```diff
@@ -62,6 +62,12 @@
     'neon': None,
 }  # T.Dicst[str, T.Optional[T.List[str]]]
 
+clang_color_args = {
+    'auto': ['-fcolor-diagnostics', '-fansi-escape-codes'],
+    'always': ['-fcolor-diagnostics', '-fansi-escape-codes'],
```

Comment:
This does not seem right. `auto` and `always` do not mean the same thing, thus it is hard to believe they can be implemented in the same way. `auto` means use colors unless stdout is not a terminal, `always` means always use colors. Using the same compiler options for both means that either you get escape sequences emitted when stdout is not a tty when `auto` is used, or you don't get them in the same situation when `always` is used.

Reply:
If I read correctly the documentation https://clang.llvm.org/docs/UsersManual.html#cmdoption-f-no-color-diagnostics this patch is not necessary. In particular, I think that enabling ansi escape codes on Windows is a bad idea.

Reply:
Why?

Reply:
Because the Windows terminal does not understand ansi escape codes.

Reply:
Hmm? The windows 10 version should.

Reply:
What to you mean by Windows 10 version of the terminal? The "Command Prompt" you get running `cmd` does not, AFAIK, and that is all is distributed with Windows 10.

Reply:
I think this is all a moot point. It's not really Meson's job to figure out whether the current prompt supports ANSI colors and then pass options to modify the compiler internals. Clang should really figure this out on its own. CLI options should just be for, well, the user telling the compiler what the user *wants*. The user wants color. The user does not care how that's implemented for the Windows API. Clang should not require users to care about this.

Reply:
From ChatGPT:

```
Rosen Penev
Does cmd.exe support ANSI escape codes?

Historically, cmd.exe on Windows has not supported ANSI escape codes, which are commonly used in Unix-like systems to control formatting and color in command-line interfaces.

However, with the release of Windows 10, support for ANSI escape codes was added to cmd.exe. To enable this support, you need to set a flag in the console properties. To do this, right-click on the title bar of the console window, select "Properties", and then click on the "Terminal" tab. Under "Terminal Colors", check the box labeled "Enable ANSI colors".

After enabling this setting, you should be able to use ANSI escape codes to control formatting and color in cmd.exe.
```

Reply:
Not listening to anything ChatGPT says, it's not a trustworthy source.

Reply:
It's answer was correct here. cmd.exe does support ANSI escape codes.

Reply:
It supports it if you explicitly enable it. It is not the default, thus it cannot be Meson default to emit ansi escape codes that will be shown as unreadable output for most users. Also, I don't think that enabling the ansi escapes codes in the Windows 10 command prompt disables the Windows Console API, thus I don't see why it is desirable to enable this option by default. The commit message says that the `-fansi-escape-codes` is required, but it does not look like this.

Reply:
Anyway, this PR came out of https://github.com/mesonbuild/wrapdb/blob/master/subprojects/packagefiles/check/meson.build#L35 , which should be default.